packages/postgress/src/MessagePickupRepositoryPg.ts

```diff
@@ -0,0 +1,616 @@
+import {
+  AddMessageOptions,
+  Agent,
+  EncryptedMessage,
+  GetAvailableMessageCountOptions,
+  injectable,
+  Logger,
+  MessagePickupEventTypes,
+  MessagePickupLiveSessionRemovedEvent,
+  MessagePickupLiveSessionSavedEvent,
+  MessagePickupRepository,
+  QueuedMessage,
+  RemoveMessagesOptions,
+  TakeFromQueueOptions,
+} from '@credo-ts/core'
+import { Pool, Client } from 'pg'
+import PGPubsub from 'pg-pubsub'
+import * as os from 'os'
+import {
+  MessageState,
+  POSTGRES_PASSWORD,
+  POSTGRES_USER,
+  POSTGRES_HOST,
+  FCM_SERVICE_BASE_URL,
+} from '../config/constants'
+import { createTableMessage, tableNameLive, createTableLive } from '../config/dbCollections'
+import { ConnectionInfo } from './interfaces'
+import { MessagePickupSession } from '@credo-ts/core/build/modules/message-pickup/MessagePickupSession'
+import axios from 'axios'
+
+@injectable()
+export class MessagePickupRepositoryPg implements MessagePickupRepository {
+  private logger?: Logger
+  public messagesCollection?: Pool
+  private agent?: Agent
+  private pubSubInstance?: PGPubsub
+  private dbChannel?: string
+  private dbListener?: boolean
+  private instanceName?: string
+  private connectionInfoCallback?: (connectionId: string) => Promise<ConnectionInfo | undefined>
+
+  public constructor(logger?: Logger) {
+    this.logger = logger
+  }
+
+  /**
+   * Initializes the service by setting up the database, event listeners, and agent.
+   *
+   * @param {Agent} agent - The agent instance to be initialized.
+   * @param {boolean} dbListener - Indicates if the pubsub listener should be active.
+   * @throws {Error} Throws an error if initialization fails.
+   */
+  public async initialize(agent: Agent, dbListener: boolean): Promise<void> {
```

Comment:
I think we can put agent and callbacks as arguments to this method. This will simplify the API and usage of this class, removing the need of separately call `setConnectionInfo` (which can be removed).

For instance, we can have:

```suggestion
  public async initialize(options: { agent: Agent, connectionInfoCallback: (connectionId: string) => Promise<ConnectionInfo | undefined>)): Promise<void> {
```

Reply:
Solved